### DISCUSSION TITLE
.pnpm-store/v3/server/socket:/requestPackage error (ECONNREFUSED).

### DISCUSSION THREAD

**Discussion by woss** [pnpm/pnpm#3402] [Q&A] [answered]
out of blue i started having this when I do `pnpm install`

info on the server

```
❯ pnpm server status
store: /home/daniel/.pnpm-store/v3
process id: 4268
remote prefix: http://unix:/home/daniel/.pnpm-store/v3/server/socket:
```

![image](https://user-images.githubusercontent.com/80440/116285599-901dd180-a78e-11eb-8ca2-4607a531a0bf.png)

Restarting the WSL didn't help, any ideas?

**Comment by woss** [comment, marked as answer]
wow, it seems i have a tendency to answer my own questions :)

I never figured out WHY but here is the destructive fix, delete ALL pnpm with store and packages

```bash
# removes the pnpm 
which pnpm | xargs rm

# remove the state and the store
rm -r ~/.pnpm-store/  ~/.pnpm-state.json

# install the pnpm again
curl -f https://get.pnpm.io/v6.js | node - add --global pnpm
```

Then it worked

**Reply by zkochan** [reply]
do you use the [pnpm server](https://pnpm.io/cli/server)?

**Reply by zkochan** [reply]
oh, silly question, the command is `pnpm server status`.

why do you need `pnpm server`?

**Reply by woss** [reply]
i have no idea why :) i was going through the docs, commands and anything that can potentially help me resolve the issue without the destructive action. at the end i never figred it out so the `destructive` action was the only option, which actually worked
